src/features/members/membersSlice.js

```diff
@@ -41,6 +50,25 @@ export const addMember = createAsyncThunk(
   },
 );
 
+export const importMembers = createAsyncThunk(
+  'members/importMembers',
+  async (args) => {
+    const { cohort, emailList } = args;
+    const client = getAuthenticatedHttpClient();
+    const baseUrl = getConfig().LMS_BASE_URL;
+    const { data } = await client.post(
+      `${baseUrl}/api/partnerships/v0/memberships/${cohort}/`,
+      snakeCaseObject(emailList.map(email => ({ email }))),
+    );
+    const newMembers = data.map(member => {
+      member.id = uniqueId('temp-');
+      return member;
+    });
+    const normalized = normalize(camelCaseObject(newMembers), memberListSchema);
```

Comment:
I see that `normalizr` has been unmaintained for a couple of years now, so I'm a bit weary of adding it as a dependency (even though it is still mentioned in the RTK docs). I'm pretty rusty on all this, but shouldn't the entity adapter be handling the normalization for us via `upsertMany`?

Reply:
I thought `upsertMany` would do it automatically as well, but it doesn't. I opted to stick with Normalizr since it was recommended and noted as being stable, but I understand the concern. Since we're really just reshaping a simple object, an 3p library may not be all that necessary at this stage. I'll do another check for something more current to replace it first though.